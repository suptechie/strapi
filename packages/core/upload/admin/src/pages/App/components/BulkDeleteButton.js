import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { Subtitle } from '@strapi/parts/Text';
import { Button } from '@strapi/parts/Button';
import { Stack } from '@strapi/parts/Stack';
import DeleteIcon from '@strapi/icons/DeleteIcon';
import { ConfirmDialog } from '@strapi/helper-plugin';
import { useBulkRemoveAsset } from '../../../hooks/useBulkRemoveAsset';
import getTrad from '../../../utils/getTrad';

export const BulkDeleteButton = ({ selectedAssets, onSuccess }) => {
  const { formatMessage } = useIntl();
  const [showConfirmDialog, setShowConfirmDialog] = useState(false);

  const { isLoading, removeAssets } = useBulkRemoveAsset();

  const handleConfirmRemove = async () => {
    await removeAssets(selectedAssets.map(({ id }) => id));
    onSuccess();
  };

  return (
    <>
      <Stack horizontal size={2} paddingBottom={5}>
        <Subtitle textColor="neutral600">
          {formatMessage(
            {
              id: getTrad('list.assets.selected'),
              defaultMessage:
                '{number, plural, =0 {No asset} one {1 asset} other {# assets}} selected',
            },
            {
              number: selectedAssets.length,
            }
          )}
        </Subtitle>
        <Button
          variant="danger-light"
          size="S"
          startIcon={<DeleteIcon />}
          onClick={() => setShowConfirmDialog(true)}
        >
          {formatMessage({ id: getTrad('control-card.delete'), defaultMessage: 'Delete' })}
        </Button>
      </Stack>

      <ConfirmDialog
        isConfirmButtonLoading={isLoading}
        isOpen={showConfirmDialog}
        onToggleDialog={() => setShowConfirmDialog(false)}
        onConfirm={handleConfirmRemove}
      />
    </>
  );
};

BulkDeleteButton.propTypes = {
  selectedAssets: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  onSuccess: PropTypes.func.isRequired,
};
