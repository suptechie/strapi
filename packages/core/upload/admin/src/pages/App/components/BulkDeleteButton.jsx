import React, { useState } from 'react';

import { ConfirmDialog } from '@strapi/admin/strapi-admin';
import { Button } from '@strapi/design-system';
import { Trash } from '@strapi/icons';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { AssetDefinition, FolderDefinition } from '../../../constants';
import { useBulkRemove } from '../../../hooks/useBulkRemove';

export const BulkDeleteButton = ({ selected, onSuccess }) => {
  const { formatMessage } = useIntl();
  const [showConfirmDialog, setShowConfirmDialog] = useState(false);
  const { remove } = useBulkRemove();

  const handleConfirmRemove = async () => {
    await remove(selected);
    onSuccess();
  };

  return (
    <>
      <Button
        variant="danger-light"
        size="S"
        startIcon={<Trash />}
        onClick={() => setShowConfirmDialog(true)}
      >
        {formatMessage({ id: 'global.delete', defaultMessage: 'Delete' })}
      </Button>

      <ConfirmDialog
        isOpen={showConfirmDialog}
        onClose={() => setShowConfirmDialog(false)}
        onConfirm={handleConfirmRemove}
      />
    </>
  );
};

BulkDeleteButton.propTypes = {
  selected: PropTypes.arrayOf(AssetDefinition, FolderDefinition).isRequired,
  onSuccess: PropTypes.func.isRequired,
};
