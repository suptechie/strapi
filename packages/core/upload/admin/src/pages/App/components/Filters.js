import React, { useState, useRef } from 'react';
import { Button } from '@strapi/parts/Button';
import { FilterPopoverURLQuery } from '@strapi/helper-plugin';
import FilterIcon from '@strapi/icons/FilterIcon';
import { useIntl } from 'react-intl';

const displayedFilters = [
  {
    name: 'created_at',
    fieldSchema: {
      type: 'date',
    },
    metadatas: { label: 'created_at' },
  },
  {
    name: 'updated_at',
    fieldSchema: {
      type: 'date',
    },
    metadatas: { label: 'updated_at' },
  },
  {
    name: 'type',
    fieldSchema: {
      type: 'enumeration',
      options: ['image', 'video', 'file'],
    },
    metadatas: { label: 'type' },
  },
];

export const Filters = () => {
  const buttonRef = useRef(null);
  const [isVisible, setVisible] = useState(false);
  const { formatMessage } = useIntl();

  const toggleFilter = () => setVisible(prev => !prev);

  return (
    <>
      <Button
        variant="tertiary"
        ref={buttonRef}
        startIcon={<FilterIcon />}
        onClick={toggleFilter}
        size="S"
      >
        {formatMessage({ id: 'app.utils.filters', defaultMessage: 'Filters' })}
      </Button>

      {isVisible && (
        <FilterPopoverURLQuery
          displayedFilters={displayedFilters}
          isVisible={isVisible}
          onToggle={toggleFilter}
          source={buttonRef}
        />
      )}
    </>
  );
};
