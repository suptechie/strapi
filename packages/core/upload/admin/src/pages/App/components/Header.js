import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { stringify } from 'qs';
import { useLocation } from 'react-router-dom';
import { useQueryParams } from '@strapi/helper-plugin';
import { HeaderLayout } from '@strapi/design-system/Layout';
import { Button } from '@strapi/design-system/Button';
import { Stack } from '@strapi/design-system/Stack';
import { Link } from '@strapi/design-system/Link';
import ArrowLeft from '@strapi/icons/ArrowLeft';
import Plus from '@strapi/icons/Plus';
import { getTrad, findRecursiveFolderMetadatas } from '../../../utils';
import { useFolderStructure } from '../../../hooks/useFolderStructure';

export const Header = ({
  assetCount,
  folderCount,
  canCreate,
  onToggleEditFolderDialog,
  onToggleUploadAssetDialog,
}) => {
  const { formatMessage } = useIntl();
  const { pathname } = useLocation();
  const [
    {
      query: { folder, ...queryParamsWithoutFolder },
    },
  ] = useQueryParams();

  const { data, isLoading } = useFolderStructure();
  const isNestedFolder = !!folder;
  const folderMetadatas = !isLoading && findRecursiveFolderMetadatas(data[0], folder);
  const backQuery = stringify(
    folderMetadatas?.parentId
      ? { ...queryParamsWithoutFolder, folder: folderMetadatas.parentId }
      : { ...queryParamsWithoutFolder },
    { encode: false }
  );

  const folderLabel =
    folderMetadatas?.currentFolderLabel &&
    (folderMetadatas.currentFolderLabel.length > 30
      ? `${folderMetadatas.currentFolderLabel.slice(0, 30)}...`
      : folderMetadatas.currentFolderLabel);

  return (
    <HeaderLayout
      title={`${formatMessage({
        id: getTrad('plugin.name'),
        defaultMessage: `Media Library`,
      })}${folderLabel ? ` - ${folderLabel}` : ''}`}
      subtitle={formatMessage(
        {
          id: getTrad('header.content.assets'),
          defaultMessage:
            '{numberFolders, plural, one {1 folder} other {# folders}} - {numberAssets, plural, one {1 asset} other {# assets}}',
        },
        { numberAssets: assetCount, numberFolders: folderCount }
      )}
      navigationAction={
        isNestedFolder && (
          <Link startIcon={<ArrowLeft />} to={`${pathname}?${backQuery}`}>
            {formatMessage({
              id: getTrad('header.actions.folder-level-up'),
              defaultMessage: 'Back',
            })}
          </Link>
        )
      }
      primaryAction={
        canCreate && (
          <Stack horizontal spacing={2}>
            <Button startIcon={<Plus />} variant="secondary" onClick={onToggleEditFolderDialog}>
              {formatMessage({
                id: getTrad('header.actions.add-folder'),
                defaultMessage: 'Add new folder',
              })}
            </Button>

            <Button startIcon={<Plus />} onClick={onToggleUploadAssetDialog}>
              {formatMessage({
                id: getTrad('header.actions.add-assets'),
                defaultMessage: 'Add new assets',
              })}
            </Button>
          </Stack>
        )
      }
    />
  );
};

Header.propTypes = {
  assetCount: PropTypes.number.isRequired,
  folderCount: PropTypes.number.isRequired,
  canCreate: PropTypes.bool.isRequired,
  onToggleEditFolderDialog: PropTypes.func.isRequired,
  onToggleUploadAssetDialog: PropTypes.func.isRequired,
};
