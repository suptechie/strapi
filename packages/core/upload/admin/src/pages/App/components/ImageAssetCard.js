import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Card,
  CardAction,
  CardAsset,
  CardBadge,
  CardBody,
  CardCheckbox,
  CardContent,
  CardHeader,
  CardTitle,
  CardSubtitle,
} from '@strapi/parts/Card';
import { IconButton } from '@strapi/parts/IconButton';
import EditIcon from '@strapi/icons/EditIcon';
import { useIntl } from 'react-intl';
import { getTrad } from '../../../utils';

const Extension = styled.span`
  text-transform: uppercase;
`;

export const ImageAssetCard = ({ name, extension, height, width, thumbnail }) => {
  const { formatMessage } = useIntl();

  return (
    <Card>
      <CardHeader>
        <CardCheckbox value />
        <CardAction position="end">
          <IconButton
            label={formatMessage({ id: getTrad('control-card.edit'), defaultMessage: 'Edit' })}
            icon={<EditIcon />}
          />
        </CardAction>
        <CardAsset src={`http://localhost:1337${thumbnail}`} />
      </CardHeader>
      <CardBody>
        <CardContent>
          <CardTitle as="h2">{name}</CardTitle>
          <CardSubtitle>
            <Extension>{extension.replace('.', '')}</Extension> - {height}âœ•{width}
          </CardSubtitle>
        </CardContent>
        <CardBadge>
          {formatMessage({ id: getTrad('settings.section.image.label'), defaultMessage: 'Image' })}
        </CardBadge>
      </CardBody>
    </Card>
  );
};

ImageAssetCard.propTypes = {
  extension: PropTypes.string.isRequired,
  height: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  thumbnail: PropTypes.string.isRequired,
};
