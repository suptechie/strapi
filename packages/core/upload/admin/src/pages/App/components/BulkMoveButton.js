import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { Button } from '@strapi/design-system/Button';
import Folder from '@strapi/icons/Folder';

import { BulkMoveDialog } from '../../../components/BulkMoveDialog';
import { AssetDefinition, FolderDefinition, FolderParentDefinition } from '../../../constants';

export const BulkMoveButton = ({ selected, onSuccess, parentFolder }) => {
  const { formatMessage } = useIntl();
  const [showConfirmDialog, setShowConfirmDialog] = useState(false);

  const handleConfirmMove = () => {
    setShowConfirmDialog(false);
    onSuccess();
  };

  return (
    <>
      <Button
        variant="secondary"
        size="S"
        startIcon={<Folder />}
        onClick={() => setShowConfirmDialog(true)}
      >
        {formatMessage({ id: 'global.move', defaultMessage: 'Move' })}
      </Button>

      {showConfirmDialog && (
        <BulkMoveDialog
          parentFolder={parentFolder}
          onClose={handleConfirmMove}
          selected={selected}
        />
      )}
    </>
  );
};

BulkMoveButton.defaultProps = {
  parentFolder: null,
};

BulkMoveButton.propTypes = {
  onSuccess: PropTypes.func.isRequired,
  parentFolder: FolderParentDefinition,
  selected: PropTypes.arrayOf(AssetDefinition, FolderDefinition).isRequired,
};
