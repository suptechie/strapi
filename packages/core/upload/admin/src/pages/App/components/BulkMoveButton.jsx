import React, { useState } from 'react';

import { Button, Modal } from '@strapi/design-system';
import { Folder } from '@strapi/icons';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { BulkMoveDialog } from '../../../components/BulkMoveDialog';
import { AssetDefinition, FolderDefinition } from '../../../constants';

export const BulkMoveButton = ({ selected, onSuccess, currentFolder }) => {
  const { formatMessage } = useIntl();
  const [showConfirmDialog, setShowConfirmDialog] = useState(false);

  const handleConfirmMove = () => {
    setShowConfirmDialog(false);
    onSuccess();
  };

  return (
    <Modal.Root open={showConfirmDialog} onOpenChange={setShowConfirmDialog}>
      <Modal.Trigger>
        <Button variant="secondary" size="S" startIcon={<Folder />}>
          {formatMessage({ id: 'global.move', defaultMessage: 'Move' })}
        </Button>
      </Modal.Trigger>
      <BulkMoveDialog
        currentFolder={currentFolder}
        onClose={handleConfirmMove}
        selected={selected}
      />
    </Modal.Root>
  );
};

BulkMoveButton.defaultProps = {
  currentFolder: undefined,
};

BulkMoveButton.propTypes = {
  onSuccess: PropTypes.func.isRequired,
  currentFolder: FolderDefinition,
  selected: PropTypes.arrayOf(AssetDefinition, FolderDefinition).isRequired,
};
