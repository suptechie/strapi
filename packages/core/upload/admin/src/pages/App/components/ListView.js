import React from 'react';
import PropTypes from 'prop-types';
import { GridLayout } from '@strapi/parts/Layout';
import { KeyboardNavigable } from '@strapi/parts/KeyboardNavigable';
import { prefixFileUrlWithBackendUrl, getFileExtension } from '@strapi/helper-plugin';
import { ImageAssetCard } from './ImageAssetCard';
import { VideoAssetCard } from './VideoAssetCard';
import { DocAssetCard } from './DocAssetCard';

export const ListView = ({ assets }) => {
  return (
    <KeyboardNavigable tagName="article">
      <GridLayout>
        {assets.map(asset => {
          if (asset.mime.includes('video')) {
            return (
              <VideoAssetCard
                id={asset.id}
                key={asset.id}
                name={asset.name}
                extension={getFileExtension(asset.ext)}
                url={prefixFileUrlWithBackendUrl(asset.url)}
                mime={asset.mime}
              />
            );
          }

          if (asset.mime.includes('image')) {
            return (
              <ImageAssetCard
                id={asset.id}
                key={asset.id}
                name={asset.name}
                extension={getFileExtension(asset.ext)}
                height={asset.height}
                width={asset.width}
                thumbnail={prefixFileUrlWithBackendUrl(asset?.formats?.thumbnail?.url || asset.url)}
              />
            );
          }

          return (
            <DocAssetCard
              id={asset.id}
              key={asset.id}
              name={asset.name}
              extension={getFileExtension(asset.ext)}
            />
          );
        })}
      </GridLayout>
    </KeyboardNavigable>
  );
};

ListView.propTypes = {
  assets: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};
