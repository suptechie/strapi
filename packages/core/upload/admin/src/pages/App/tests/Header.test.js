import React from 'react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { QueryClientProvider, QueryClient } from 'react-query';
import { render } from '@testing-library/react';
import { useQueryParams } from '@strapi/helper-plugin';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

import { Header } from '../components/Header';
import { useFolderStructure } from '../../../hooks/useFolderStructure';

jest.mock('../../../hooks/useFolderStructure');

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useQueryParams: jest.fn().mockReturnValue([{ query: {}, rawQuery: '' }, jest.fn()]),
}));

const setup = props => {
  const withDefaults = {
    ...props,
    assetCount: 2,
    folderCount: 2,
    canCreate: true,
    onToggleEditFolderDialog: jest.fn(),
    onToggleUploadAssetDialog: jest.fn(),
  };

  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en">
          <MemoryRouter>
            <Header {...withDefaults} />
          </MemoryRouter>
        </IntlProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

describe('Header', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('truncates long folder lavels', () => {
    useQueryParams.mockReturnValueOnce([{ rawQuery: '', query: { folder: 2 } }, jest.fn()]);
    useFolderStructure.mockReturnValueOnce({
      isLoading: false,
      error: null,
      data: [
        {
          value: null,
          label: 'Media Library',
          children: [
            {
              value: 1,
              label: 'Cats',
              children: [
                {
                  value: 2,
                  label: 'The length of this label exceeds the maximum',
                  children: [],
                },
              ],
            },
          ],
        },
      ],
    });

    const { queryByText } = setup();
    expect(queryByText('Media Library - The length of this label excee...')).toBeInTheDocument();
  });

  test('does not render a back button at the root level of the media library', () => {
    const { queryByText } = setup();

    expect(queryByText('Back')).not.toBeInTheDocument();
  });

  test('does render a back button at a nested level of the media library', () => {
    useQueryParams.mockReturnValueOnce([{ rawQuery: '', query: { folder: 2 } }, jest.fn()]);
    const { queryByText } = setup();

    expect(queryByText('Back')).toBeInTheDocument();
  });
});
