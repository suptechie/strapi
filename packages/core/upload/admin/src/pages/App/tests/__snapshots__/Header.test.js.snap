// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
.c19 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c17 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c14 {
  padding-right: 8px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c11 svg {
  height: 12px;
  width: 12px;
}

.c11 svg > g,
.c11 svg path {
  fill: #ffffff;
}

.c11[aria-disabled='true'] {
  pointer-events: none;
}

.c11:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c11:focus-visible {
  outline: none;
}

.c11:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c15 {
  height: 100%;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #4945ff;
  border: 1px solid #4945ff;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c12 .c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 .c16 {
  color: #ffffff;
}

.c12[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c12[aria-disabled='true'] .c16 {
  color: #666687;
}

.c12[aria-disabled='true'] svg > g,
.c12[aria-disabled='true'] svg path {
  fill: #666687;
}

.c12[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c12[aria-disabled='true']:active .c16 {
  color: #666687;
}

.c12[aria-disabled='true']:active svg > g,
.c12[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c12:hover {
  background-color: #ffffff;
}

.c12:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c12:active .c16 {
  color: #4945ff;
}

.c12:active svg > g,
.c12:active svg path {
  fill: #4945ff;
}

.c12 .c16 {
  color: #271fe0;
}

.c12 svg > g,
.c12 svg path {
  fill: #271fe0;
}

.c18 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #4945ff;
  border: 1px solid #4945ff;
}

.c18 .c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 .c16 {
  color: #ffffff;
}

.c18[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c18[aria-disabled='true'] .c16 {
  color: #666687;
}

.c18[aria-disabled='true'] svg > g,
.c18[aria-disabled='true'] svg path {
  fill: #666687;
}

.c18[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c18[aria-disabled='true']:active .c16 {
  color: #666687;
}

.c18[aria-disabled='true']:active svg > g,
.c18[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c18:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c18:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c18 svg > g,
.c18 svg path {
  fill: #ffffff;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 > * {
  margin-left: 0;
  margin-right: 0;
}

.c10 > * + * {
  margin-left: 8px;
}

.c0 {
  background: #f6f6f9;
  padding-top: 24px;
  padding-right: 56px;
  padding-bottom: 40px;
  padding-left: 56px;
}

.c1 {
  padding-bottom: 8px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  color: #32324d;
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
}

.c5 {
  color: #4945ff;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c3 {
  padding-right: 8px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  outline: none;
}

.c2 svg path {
  fill: #4945ff;
}

.c2 svg {
  font-size: 0.625rem;
}

.c2:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c2:focus-visible {
  outline: none;
}

.c2:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    style="height: 0px;"
  >
    <div
      class="c0"
      data-strapi-header="true"
    >
      <div
        class="c1"
      >
        <a
          aria-current="page"
          class="c2 active"
          href="/?folder=2"
        >
          <span
            aria-hidden="true"
            class="c3 c4"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 13.3a.2.2 0 01-.2.2H5.74l8.239 8.239a.2.2 0 010 .282L12.14 23.86a.2.2 0 01-.282 0L.14 12.14a.2.2 0 010-.282L11.86.14a.2.2 0 01.282 0L13.98 1.98a.2.2 0 010 .282L5.74 10.5H23.8c.11 0 .2.09.2.2v2.6z"
                fill="#212134"
              />
            </svg>
          </span>
          <span
            class="c5"
          >
            Back
          </span>
        </a>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <h1
            class="c8"
          >
            Media Library
          </h1>
        </div>
        <div
          class="c9 c10"
          spacing="2"
        >
          <button
            aria-disabled="false"
            class="c11 c12"
            type="button"
          >
            <div
              aria-hidden="true"
              class="c13 c14 c15"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                  fill="#212134"
                />
              </svg>
            </div>
            <span
              class="c16 c17"
            >
              Add new folder
            </span>
          </button>
          <button
            aria-disabled="false"
            class="c11 c18"
            type="button"
          >
            <div
              aria-hidden="true"
              class="c13 c14 c15"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                  fill="#212134"
                />
              </svg>
            </div>
            <span
              class="c16 c17"
            >
              Add new assets
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c19"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
